lightning/src/ln/channelmanager.rs

```diff
@@ -12727,15 +12473,7 @@ where
 		_responder: Option<Responder>
 	) -> Option<(ReleaseHeldHtlc, ResponseInstruction)> {
 		#[cfg(async_payments)] {
-			match _context {
-				AsyncPaymentsContext::InboundPayment { nonce, hmac, path_absolute_expiry } => {
-					if let Err(()) = signer::verify_held_htlc_available_context(
-						nonce, hmac, &self.inbound_payment_key
-					) { return None }
-					if self.duration_since_epoch() > path_absolute_expiry { return None }
-				},
-				_ => return None
-			}
+			if self.flow.verify_async_context(_context).is_err() { return None }
```

Comment:
This may be unlikely, but currently if the context is a valid `AsyncPaymentsContext::OutboundPayment`, we will pass the check here even though we should fail. So I think we might want separate `verify_async_{inbound,outbound}_payment` methods to make sure we have the right context here and below. Let me know if that doesn't make sense.